doc/channel-plugin.md

```diff
@@ -0,0 +1,58 @@
+## Channel Plugin
+
+Channel plugins are processes that run independently of each other.
+
+Channel plugin uses stdin and stdout as json format to communicate. The request and response structure is inspired by
```

Comment:
I'd write the first sentence something more like this:

> Channel plugins receive JSON-formatted requests on stdin and reply with JSON-formatted responses on stdout.

It should also be made clear that these are long-running processes and receive many requests over their lifetime.